examples/vision/segformer.py

```diff
@@ -0,0 +1,288 @@
+"""
+We can verify the shapes of the input images and their segmentation maps.
+"""
+
+print(train_ds.element_spec)
+
+"""
+## Visualize dataset
+"""
+
+import matplotlib.pyplot as plt
+
+
+def display(display_list):
+    plt.figure(figsize=(15, 15))
+
+    title = ["Input Image", "True Mask", "Predicted Mask"]
+
+    for i in range(len(display_list)):
+        plt.subplot(1, len(display_list), i + 1)
+        plt.title(title[i])
+        plt.imshow(tf.keras.utils.array_to_img(display_list[i]))
+        plt.axis("off")
+    plt.show()
+
+
+for samples in train_ds.take(2):
+    sample_image, sample_mask = samples["pixel_values"][0], samples["labels"][0]
+    sample_image = tf.transpose(sample_image, (1, 2, 0))
+    sample_mask = tf.expand_dims(sample_mask, -1)
+    display([sample_image, sample_mask])
+
+"""
+## Load a pre-trained SegFormer checkpoint
+
+We now load a pre-trained SegFormer model variant from Hugging Face Transformers. The
+SegFormer model comes in different variants dubbed as **MiT-B0** to **MiT-B5**. You can
+find these checkpoints
+[here](https://huggingface.co/models?pipeline_tag=image-segmentation&sort=downloads&search=segformer).
+We'll load the smallest variant Mix-B0 which produces a good trade-off
+between inference efficiency and predictive performance.
+"""
+
+from transformers import TFSegformerForSemanticSegmentation
+
+model_checkpoint = "nvidia/mit-b0"
+id2label = {0: "outer", 1: "inner", 2: "border"}
+label2id = {label: id for id, label in id2label.items()}
+num_labels = len(id2label)
+model = TFSegformerForSemanticSegmentation.from_pretrained(
+    model_checkpoint,
+    num_labels=num_labels,
+    id2label=id2label,
+    label2id=label2id,
+    ignore_mismatched_sizes=True,
+)
+
+"""
+The warning is telling us that we're throwing away some weights and newly initializing
+some others. Don't panic! This is absolutely normal. Since we're using a custom dataset
+which has a different set of semantic class labels than the pre-training dataset,
+[`TFSegformerForSemanticSegmentation`](https://huggingface.co/docs/transformers/model_doc/segformer#transformers.TFSegformerForSemanticSegmentation)
+is initializing a new decoder head.  
+
+We can now initialize an optimizer and compile the model with it. 
+"""
+
+"""
+## Compile the model
+"""
+
+lr = 0.00006
+optimizer = tf.keras.optimizers.Adam(learning_rate=lr)
+model.compile(optimizer=optimizer)
+
+"""
+Notice that we are not using any loss function for compiling the model. This is because
+the forward pass of the model
+[implements](https://github.com/huggingface/transformers/blob/820c46a707ddd033975bc3b0549eea200e64c7da/src/transformers/models/segformer/modeling_tf_segformer.py#L873)
+the loss computation part when we provide labels alongside the input images. After
+computing the loss, the model returned a structured `dataclass` object which is
+then used to guide the training process.
+
+With the compiled, we can proceed and call `fit()` on it to begin the fine-tuning
```

Comment:
nit: maybe? Or "With it compiled"? 

```suggestion
With the compiled model, we can proceed and call `fit()` on it to begin the fine-tuning
```